libs/langchain/langchain/retrievers/multi_vector.py

```diff
@@ -27,12 +28,40 @@ class MultiVectorRetriever(BaseRetriever):
     and their embedding vectors"""
     docstore: BaseStore[str, Document]
     """The storage layer for the parent documents"""
-    id_key: str = "doc_id"
-    search_kwargs: dict = Field(default_factory=dict)
+    id_key: str
+    search_kwargs: dict
     """Keyword arguments to pass to the search function."""
-    search_type: SearchType = SearchType.similarity
+    search_type: SearchType
     """Type of search to perform (similarity / mmr)"""
 
+    def __init__(
+        self,
+        *,
+        vectorstore: VectorStore,
+        docstore: Optional[BaseStore[str, Document]] = None,
+        base_store: Optional[BaseStore[str, bytes]] = None,
+        id_key: str = "doc_id",
+        search_kwargs: dict = {},
+        search_type: SearchType = SearchType.similarity,
+    ):
+        if base_store is not None:
+            docstore = create_kv_docstore(base_store)
+        elif docstore is not None:
+            warnings.warn(
```

Comment:
Let's not deprecate this. We don't know how users created the implementation (it may not have been through create_kv_docstore) -- i.e., could be through inheritance of basestore

Reply:
We haven't documented at all so I'd be surprised if anyone figured it out but yeah I can remove